mathics/core/parser/tokeniser.py

```diff
@@ -230,7 +233,7 @@
     '<': ['Get', 'StringJoin', 'LessEqual', 'Less'],
     '=': ['SameQ', 'UnsameQ', 'Equal', 'Unset', 'Set'],
     '>': ['PutAppend', 'Put', 'GreaterEqual', 'Greater'],
-    '?': ['PatternTest'],
+    '?': ['Information', 'Definition','PatternTest'],
```

Comment:
The tokeniser does not distinguish between prefix and binary operators at all, that's the parsers job. For example, `1 + 2` will tokenise as a sequence `[Number, Plus, Number]` but `+3` would just be `[Plus Number]`.

The parser (in `core/parser/parser.py`) knows whether it's at the start of an expression or part-way through.

At the start of an expression it calls `parse_p` for parsing prefix expressions. It looks for a manual rule of the form `p_TOKENNAME` (in this case `p_Plus`). Alternatively it looks for the token in `prefix_ops` (defined in `core/parser/operators.py`) which is handled directly. If neither case applies it emits a syntax message.

In the current master `+3` finds `p_Plus` but `?Sin` can't find either a `p_PatternTest` nor find it in the `prefix_ops` dict so it emits the syntax error message that we currently see.

Does that make sense?